plugins/InspectionGadgets/InspectionGadgetsAnalysis/src/com/siyeh/ig/controlflow/SimplifiableEqualsExpressionInspection.java

```diff
@@ -184,7 +185,7 @@ public void visitPolyadicExpression(PsiPolyadicExpression expression) {
         if (!isEqualsConstant(rhs, variable)) {
           return;
         }
-        registerError(lhs, getMethodName((PsiMethodCallExpression)rhs));
+        registerError(lhs, ProblemHighlightType.LIKE_UNUSED_SYMBOL, getMethodName((PsiMethodCallExpression)rhs));
```

Comment:
it's wrong just to delete the `lhs`, one needs changes on the right side (probably). Thus, I'd avoid the change here

Reply:
I will handle this more complex case in a further PR. I rollback.